src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -382,11 +382,9 @@ private void walk(DetailNode root) {
                 }
 
                 toVisit = JavadocUtil.getNextSibling(curNode);
-                if (toVisit == null) {
-                    curNode = curNode.getParent();
-                    if (curNode != null) {
-                        waitsForProcessing = shouldBeProcessed(curNode);
-                    }
+                curNode = curNode.getParent();
```

Comment:
I can not believe it can be ok, we need to pass through all siblings before pop to parent

Reply:
> we need to pass through all siblings before pop to parent

@romani  yes, indeed that is happening proven by analysis and debugging. can you please explain why you think it is not working

Reply:
@romani, I agree with @mahfouz72's explanation. This change shouldn't affect the logic as the code semantically the same as it used to be before the change. If you check the outer `while` loop and how `curNode` is reset by `toVisit` after the inner `while` loop, it should be more clear.

Here's code before the change:
```java
        while (curNode != null) {
            boolean waitsForProcessing = shouldBeProcessed(curNode);

            if (waitsForProcessing) {
                visitJavadocToken(curNode);
            }
            DetailNode toVisit = JavadocUtil.getFirstChild(curNode);
            while (curNode != null && toVisit == null) {
                if (waitsForProcessing) {
                    leaveJavadocToken(curNode);
                }

                toVisit = JavadocUtil.getNextSibling(curNode);
                if (toVisit == null) {
                    curNode = curNode.getParent();
                    if (curNode != null) {
                        waitsForProcessing = shouldBeProcessed(curNode);
                    }
                }
            }
            curNode = toVisit;
        }
``` 
By removing `if (toVisit == null)` we end up with the following:

1. This logic is now executed in any case
```java
                  curNode = curNode.getParent();
                  if (curNode != null) {
                      waitsForProcessing = shouldBeProcessed(curNode);
                  }
```
2. When `toVisit != null` we exit the inner `while` loop, where:
2.1. `curNode` is reset due to`curNode = toVisit`, so we don't actually use new value obtained from `curNode = curNode.getParent()`.
2.2. At the beginning of the next outer `while` loop iteration `waitsForProcessing` is reset due to `boolean waitsForProcessing = shouldBeProcessed(curNode)`, so we don't actually use previous value obtained from `waitsForProcessing = shouldBeProcessed(curNode)` in the innter loop.